### PR TITLE
Extract Block Transforms into its own handbook page

### PR DESCRIPTION
**Pull request by oandregal**
Part of https://github.com/WordPress/gutenberg/issues/20161

This PR extracts and extends the existing Block Transforms API docs into a sub-section of "Block API Reference".

- Adds and/or normalizes contents for every type of transformation currently available: `block`, `enter`, `files`, `prefix`, `raw`, `shortcode`.
- Refactors the existing transforms section within the block registration to point to the new page.
- Adds a new page to the handbook.

Compare: [new content](https://github.com/WordPress/gutenberg/blob/add/block-transforms-docs/docs/designers-developers/developers/block-api/block-transforms.md), [old content](https://github.com/WordPress/gutenberg/blob/master/docs/designers-developers/developers/block-api/block-registration.md#transforms-optional).

### CODE REVIEW

File: docs/designers-developers/developers/block-api/block-transforms.md [hunk 1]
```diff
@@ -0,0 +1,462 @@
...
+
+- **type** _(string)_: the value `raw`.
+- **transform** _(function, optional)_: a callback that receives the node being processed. It should return a block object or an array of block objects.
+- **schema** _(object|function, optional)_: it defines the attributes and children of the node that will be preserved on paste, according to its [HTML content model](https://html.spec.whatwg.org/multipage/dom.html#content-models). Take a look at [pasteHandler](/packages/blocks/README.md#pasteHandler) for more info.
+- **selector** _(string, optional)_: a CSS selector string to determine whether the element matches according to the [element.matches](https://developer.mozilla.org/en-US/docs/Web/API/Element/matches) method. The transform won't be executed if the element doesn't match. This is a shorthand and alternative to using `isMatch`, which, if present, will take precedence.
+- **isMatch** _(function, optional)_: a callback that receives the node being processed and should return a boolean. Returning `false` from this function will prevent the transform from being applieddisplayed as an option to the user.
```
Comment (ellatrix) [inline docs/designers-developers/developers/block-api/block-transforms.md:321]:
```suggestion
- **isMatch** _(function, optional)_: a callback that receives the node being processed and should return a boolean. Returning `false` from this function will prevent the transform from being applied displayed as an option to the user.
```

**Review by ellatrix — APPROVED** [review]

File: docs/designers-developers/developers/block-api/block-transforms.md (hunk 1 above)
Comment (nosolosw) [reply docs/designers-developers/developers/block-api/block-transforms.md]: Fixed at https://github.com/WordPress/gutenberg/pull/21734/commits/331bb987369155a14559cfb640d1514425adad8c together with other `isMatch` fixes that suffered much from copy/pasting!

**Review by mkaz — APPROVED** [review]
👍 Good stuff Andres. Thanks for contributing and helping keep the docs organized.

We may want to cherry-pick to branch if possible to update Block Editor handbook which is pinned to `wp/5.4` branch. Though it might potentially be tricky depending on other doc changes.

What we really need is multiple versions of docs published and be able to toggle between branches.

**Comment by nosolosw** [comment]
@mkaz In doing some archeology I found the [old ticket for sub-components README](https://github.com/WordPress/gutenberg/issues/15107)!

**Comment by nosolosw** [comment]
[Follow-up about how to update handbook's contents](https://github.com/WordPress/gutenberg/issues/21858).

**Pull request merged**